test/cases/uri_gid_test.rb

```diff
@@ -55,7 +55,8 @@ class URI::GIDValidationTest < ActiveSupport::TestCase
   end
 
   test 'missing model id' do
-    assert_invalid_component 'gid://bcx/Person'
+    err = assert_raise(URI::GID::MissingModelId) { URI::GID.parse('gid://bcx/Person') }
+    assert_match /Unable to create a global id for Person/, err.message
```

Comment:
I added back the error message check since it would have caught the last issue you brought up.